demosplan/DemosPlanCoreBundle/Logic/StatementSynchronizer.php

```diff
@@ -250,24 +259,47 @@ private function copyFileContainersBetweenStatements(
      *
      * @throws Exception
      */
-    private function copyFileContainersToStatement(array $fileContainers, Statement $targetStatement, bool $createLinks): array
-    {
+    private function copyFileContainersToStatement(
+        array $fileContainers,
+        Statement $targetStatement
+    ): array {
         $fileContainerCopies = [];
         foreach ($fileContainers as $fileContainer) {
             $newFileContainer = $this->statementRepository->copyFileContainer($fileContainer, $targetStatement);
             $fileViolations = $this->validator->validate($newFileContainer);
             if (0 !== $fileViolations->count()) {
                 throw ViolationsException::fromConstraintViolationList($fileViolations);
             }
-            if ($createLinks) {
-                $this->createAndPersistLink($fileContainer, $newFileContainer);
-            }
+            $this->createAndPersistLink($fileContainer, $newFileContainer);
             $fileContainerCopies[] = $newFileContainer;
         }
 
-        $targetStatement->setFiles(array_map(static fn (FileContainer $fileContainer): string => $fileContainer->getFileString(), $fileContainerCopies));
+        $targetStatement->setFiles(
+            array_map(
+                static fn (FileContainer $fileContainer): string => $fileContainer->getFileString(),
+                $fileContainerCopies
+            )
+        );
+
+        return $fileContainerCopies;
+    }
+
+    /**
+     * @param array<int, FileContainer> $originalfileContainers
+     *
+     * @throws Exception
+     */
+    public function cloneFileContainersToStatement(
```

Comment:
maybe a comment would be helpful why a distinct "clone" method is needed and what is meant by cloning and in which case this should be favored against copying

Reply:
done in adfe67f219cae62f6f425f326db47dacd9aa0180